### PR TITLE
Migrate rustc_mir_dataflow to diagnostic structs

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by michaelwoerister** [comment]
r? diagnostics

File: compiler/rustc_error_messages/locales/en-US/rustc_mir_dataflow.ftl [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    `{$name}` requires an argument
+
+rustc_mir_dataflow_stop_after_dataflow_ended_compilation =
+    stop_after_dataflow ended compilation
+
+rustc_mir_dataflow_peek_must_be_place_or_ref_place =
+    rustc_peek: argument expression must be either `place` or `&place`
+
+rustc_mir_dataflow_peek_must_be_not_temporary =
+    dataflow::sanity_check cannot feed a non-temp to rustc_peek.
```
Comment (JohnTitor) [inline compiler/rustc_error_messages/locales/en-US/rustc_mir_dataflow.ftl:20]: Pre-existing but could you remove the below period? IIRC we generally don't add it in diagnostics.

```suggestion
    dataflow::sanity_check cannot feed a non-temp to rustc_peek
```

File: compiler/rustc_error_messages/src/lib.rs [hunk 2]
```diff
@@ -43,6 +43,7 @@ fluent_messages! {
     passes => "../locales/en-US/passes.ftl",
     privacy => "../locales/en-US/privacy.ftl",
     typeck => "../locales/en-US/typeck.ftl",
+    rustc_mir_dataflow => "../locales/en-US/rustc_mir_dataflow.ftl",
```
Comment (davidtwco) [inline compiler/rustc_error_messages/src/lib.rs:46]:
```suggestion
    mir_dataflow => "../locales/en-US/rustc_mir_dataflow.ftl",
```

We can just call this `mir_dataflow` everywhere.

**Review by davidtwco — CHANGES_REQUESTED** [review]
LGTM, one small change then r=me

**Review by davidtwco — APPROVED** [review]

File: compiler/rustc_mir_dataflow/Cargo.toml [hunk 3]
```diff
@@ -13,10 +13,13 @@ smallvec = { version = "1.8.1", features = ["union", "may_dangle"] }
...
 rustc_ast = { path = "../rustc_ast" }
 rustc_data_structures = { path = "../rustc_data_structures" }
+rustc_errors = { path = "../rustc_errors" }
 rustc_graphviz = { path = "../rustc_graphviz" }
 rustc_hir = { path = "../rustc_hir" }
 rustc_index = { path = "../rustc_index" }
+rustc_macros = { path = "../rustc_macros" }
 rustc_middle = { path = "../rustc_middle" }
 rustc_serialize = { path = "../rustc_serialize" }
+rustc_session = { path = "../rustc_session" }
```
Comment (klensy) [inline compiler/rustc_mir_dataflow/Cargo.toml:23]: Don't see where `rustc_session` used :-(

**Pull request merged**